pkg/controller/job/job_controller.go

```diff
@@ -1871,7 +1871,10 @@ func countReadyPods(pods []*v1.Pod) int32 {
 // PodReplacementPolicy controls when we recreate pods if they are marked as terminating
 // Failed means that we recreate only once the pod has terminated.
 func onlyReplaceFailedPods(job *batch.Job) bool {
-	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) && *job.Spec.PodReplacementPolicy == batch.Failed {
+	// We check both PodReplacementPolicy for nil and failed because
+	// there could be a possibility where PodReplacementPolicy was not defaulted to TerminatingOrFailed or Failed
+	// This fixes a potential segmentation fault where apiserver didn't default this field.
```

Comment:
Maybe with the sentence above this one is no longer needed.

Reply:
dropped.